File: benchmark/julia/task6_fileio.jl
function count_words(input_file, output_file)
    text = read(input_file, String)
    words = split(text)
    counts = Dict{String, Int}()

    for word in words
        counts[word] = get(counts, word, 0) + 1
    end

    open(output_file, "w") do f
        for (word, count) in counts
            println(f, "$word: $count")
        end
    end
end

File: benchmark/ruby/task6_fileio.rb
def count_words(input_file, output_file)
  text = File.read(input_file)
  counts = Hash.new(0)
  text.split.each { |word| counts[word] += 1 }

  File.open(output_file, 'w') do |f|
    counts.each { |word, count| f.puts "#{word}: #{count}" }
  end
end
